src/proto-banner1.c

```diff
@@ -127,10 +129,10 @@ banner1_parse(
                         banner1->smack,
                         &tcb_state->state,
                         px, &offset, (unsigned)length);
-        if (x != SMACK_NOT_FOUND)
-            proto = patterns[x].id;
-        else
+        if (x == SMACK_NOT_FOUND)
             proto = 0xFFFFFFFF;
+        else
+            proto = patterns[x].id;
```

Comment:
What's the point of this change? Is it related to this PR?

Reply:
Remove a double negative I can drop it.